fe/fe-core/src/main/java/com/starrocks/catalog/ForeignKeyConstraint.java

```diff
@@ -165,12 +167,23 @@ public static List<ForeignKeyConstraint> parse(String foreignKeyConstraintDescs)
         return foreignKeyConstraints;
     }
 
-    private static BaseTableInfo getTableBaseInfo(String catalog, String db, String table, String tableIdentifier) {
+    private static BaseTableInfo getTableBaseInfo(String catalogName, String db, String table, String tableIdentifier) {
         BaseTableInfo baseTableInfo;
-        if (catalog.equals(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME)) {
-            baseTableInfo = new BaseTableInfo(Long.parseLong(db), Long.parseLong(table));
+        if (catalogName.equals(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME)) {
+            Preconditions.checkArgument(NumberUtils.isNumber(db),
+                    "BaseTableInfo db %s should be dbId for internal catalog", db);
+            Preconditions.checkArgument(NumberUtils.isNumber(table),
+                    "BaseTableInfo table %s should be tableId for internal catalog", table);
+            long dbId = Long.parseLong(db);
+            long tableId = Long.parseLong(table);
+            Database database = GlobalStateMgr.getCurrentState().getDb(dbId);
+            Preconditions.checkArgument(database != null, "BaseInfo's db %s should not null", dbId);
+            Table baseTable = database.getTable(tableId);
+            Preconditions.checkArgument(baseTable != null,
+                    "BaseInfo's baseTable %s should not null", tableId);
+            baseTableInfo = new BaseTableInfo(dbId, database.getFullName(), baseTable.getName(), tableId);
```

Comment:
what will happen if these checks throw exception?

Reply:
ignore the constraints.
```

    public TableProperty buildConstraint() {
        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_UNIQUE_CONSTRAINT)) {
            try {
                uniqueConstraints = UniqueConstraint.parse(
                        properties.getOrDefault(PropertyAnalyzer.PROPERTIES_UNIQUE_CONSTRAINT, ""));
            } catch (Exception e) {
                LOG.warn("Failed to parse unique constraints, ignore this unique constraint", e);
            }
        }

        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_FOREIGN_KEY_CONSTRAINT)) {
            try {
                foreignKeyConstraints = ForeignKeyConstraint.parse(
                        properties.getOrDefault(PropertyAnalyzer.PROPERTIES_FOREIGN_KEY_CONSTRAINT, ""));
            } catch (Exception e) {
                LOG.warn("Failed to parse foreign key constraints, ignore this foreign key constraints", e);
            }
        }
        return this;
    }
```